src/common/ini_handler.c

```diff
@@ -40,3 +48,165 @@ uint8_t ini_load_file(ini_handler handler, void *user_struct) {
     }
     return 1;
 }
+
+static void _change_any_to_static(_networkconfig_t * config) {
+    if (netif_is_up(&eth0)) {
+        netifapi_netif_set_down(&eth0);
+    }
+    config->lan_flag |= LAN_EEFLG_TYPE;
+    eeprom_set_var(EEVAR_LAN_FLAG, variant8_ui8(config->lan_flag));
+
+    netifapi_netif_set_addr(&eth0,
+        (const ip4_addr_t *)&(config->lan.addr_ip4),
+        (const ip4_addr_t *)&(config->lan.msk_ip4),
+        (const ip4_addr_t *)&(config->lan.gw_ip4)
+        );
+#ifdef ENABLE_DNS
+    dns_setserver(0, (const ip4_addr_t *)&(config->dns1_ip4));
+    dns_setserver(1, (const ip4_addr_t *)&(config->dns2_ip4));
+#endif
+    if (netif_is_link_up(&eth0) && !(config->lan_flag & LAN_EEFLG_ONOFF)) {
+        netifapi_netif_set_up(&eth0);
+    }
+}
+
+static void _change_static_to_dhcp(uint8_t lan_flg) {
+    if (netif_is_up(&eth0)) {
```

Comment:
this should not be here

Reply:
I agree, there should be a separate file handling all the netif stuff. Until now, the whole solution was in screen_lan_settings, so I had to move* it from GUI.. 

* copy and paste as you said no touching of screen_lan_settings